components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/push/notification/mgt/task/PushNotificationSchedulerTaskTest.java

```diff
@@ -0,0 +1,89 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.device.mgt.core.push.notification.mgt.task;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.mockito.Mockito;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.device.mgt.common.DeviceManagementException;
+import org.wso2.carbon.device.mgt.common.operation.mgt.OperationManagementException;
+import org.wso2.carbon.device.mgt.core.common.BaseDeviceManagementTest;
+import org.wso2.carbon.device.mgt.core.config.DeviceConfigurationManager;
+import org.wso2.carbon.device.mgt.core.dto.operation.mgt.Operation;
+import org.wso2.carbon.device.mgt.core.internal.DeviceManagementDataHolder;
+import org.wso2.carbon.device.mgt.core.internal.DeviceManagementServiceComponent;
+import org.wso2.carbon.device.mgt.core.operation.TestNotificationStrategy;
+import org.wso2.carbon.device.mgt.core.operation.mgt.OperationMapping;
+import org.wso2.carbon.device.mgt.core.operation.mgt.dao.OperationDAO;
+import org.wso2.carbon.device.mgt.core.operation.mgt.dao.OperationManagementDAOException;
+import org.wso2.carbon.device.mgt.core.operation.mgt.dao.OperationManagementDAOFactory;
+import org.wso2.carbon.device.mgt.core.service.DeviceManagementProviderService;
+import org.wso2.carbon.device.mgt.core.service.DeviceManagementProviderServiceImpl;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+public class PushNotificationSchedulerTaskTest extends BaseDeviceManagementTest {
+    private static final Log log = LogFactory.getLog(PushNotificationSchedulerTask.class);
+    private DeviceManagementProviderService deviceMgtProviderService;
+    private PushNotificationSchedulerTask pushNotificationSchedulerTask;
+    private OperationDAO operationDAO;
+
+    @BeforeClass
+    public void init() throws DeviceManagementException, RegistryException {
+        DeviceConfigurationManager.getInstance().initConfig();
+        log.info("Initializing Push Notification Scheduler Test Class");
+        DeviceManagementServiceComponent.notifyStartupListeners();
+        this.deviceMgtProviderService = Mockito.mock(DeviceManagementProviderServiceImpl.class, Mockito.CALLS_REAL_METHODS);
+        DeviceManagementDataHolder.getInstance().setDeviceManagementProvider(this.deviceMgtProviderService);
+        this.operationDAO = OperationManagementDAOFactory.getOperationDAO();
+        this.pushNotificationSchedulerTask = new PushNotificationSchedulerTask();
+    }
+
+    @Test
+    public void testPushNotificationScheduler() {
+        try {
+            log.debug("Attempting to execute push notification task scheduler");
```

Comment:
Do we need debug logs in test case? Since this is a test case, isn't better to have info logs?